src/gluonts/model/forecast_generator.py

```diff
@@ -223,3 +154,16 @@ def __call__(
                     info=batch["info"][i] if "info" in batch else None,
                 )
             assert i + 1 == len(batch["forecast_start"])
+
+
+def DistributionForecastGenerator(*args, **kwargs):
```

Comment:
I wonder, maybe something like this https://www.python.org/dev/peps/pep-0549/ is a better way to do the deprecation warning.

Reply:
New proposal: We just re-import moved classes at their old locations and add an issue to deprecate these later.

That way we can merge this soon and thing about deprecation later.

Reply:
> New proposal: We just re-import moved classes at their old locations and add an issue to deprecate these later.

I second this

Reply:
How would you do that concretely? Just importing the classes from the new location would introduce circular imports, since the mx classes depend on the generic classes.

Reply:
I just tried it and it seems to be working: at the bottom of `src/gluonts/model/forecast_generator.py` I put

```python
from gluonts.mx.model.forecast_generator import DistributionForecastGenerator
```

and then I can do

```python
>>> from gluonts.model.forecast_generator import DistributionForecastGenerator
>>> DistributionForecastGenerator
<class 'gluonts.mx.model.forecast_generator.DistributionForecastGenerator'>
```

Similarly for `src/gluonts/model/predictor.py`, if I replace your definitions with

```python
from gluonts.mx.model.predictor import GluonPredictor, SymbolBlockPredictor, RepresentableBlockPredictor
```

I think the circular import issue shows up if instead you move these imports to the top of the modules.

Reply:
Ah! Thanks :)